docs/proxies.md

```diff
@@ -51,6 +51,35 @@ proxy = httpcore.HTTPProxy(
 )
 ```
 
+## Proxy Connection mode
+
+There are two types of HTTP proxy:
+
+1. Forwading (using absolute-url)
+2. Tunneling (using `CONNECT` method)
+
+By default we forwarding plain http requests and tunneling https requests.
+You can change this behavior with `mode: httpcore.ProxyMode` parameter:
+```py
+import httpcore
+import base64
+
+proxy = httpcore.HTTPProxy(
+    proxy_url="http://127.0.0.1:8080/",
+    mode=httpcore.ProxyMode.HTTP_TUNNEL,  # Forces HTTP requests also use Tunneling
+)
+```
+`ProxyMode` is a flag parameter, also you can reverse current default behavior:
+```py
+import httpcore
+import base64
+
+proxy = httpcore.HTTPProxy(
+    proxy_url="http://127.0.0.1:8080/",
+    mode=httpcore.ProxyMode.HTTP_TUNNEL | httpcore.ProxyMode.HTTPS_FORWARD,
+)
+```
+(WARN: `ProxyMode.HTTPS_FORWARD` is unsafe, use it when you trust proxy server)
```

Comment:
how do other texts phrase this? can we explain the implications to readers more clearly?

Reply:
perhaps here needs fully explain that what is forwarding and tunneling proxy.